packages/@vue/cli-ui-addon-webpack/.eslintrc.js

```diff
@@ -9,5 +9,11 @@ module.exports = {
     mapSharedData: false,
     Vue: false,
     name: 'off'
+  },
+  parserOptions: {
+    parser: '@babel/eslint-parser',
+    babelOptions: {
+      cwd: __dirname
+    }
```

Comment:
`@babel/eslint-parser` needs load babelConfig. 

Since we needn't merge sub-package `babelrc` with root babel.config.js(we don't have root babel.config.js). Just set `cwd`/[`root`](https://babeljs.io/docs/en/options#root)  to `'package/@vu/cli-ui-addon-webpack'`, `@babel/eslint-parser` will load `babelrc` from `cwd: __dirname`  successfully when executing `yarn lint` in project root directory. 

If we need merge `babelrc` with root `babel.config.js` in monorepo, we should set [`babelrcRoot`](https://babeljs.io/docs/en/config-files#subpackage-babelrcjson-files).

Reply:
If we rename `.babelrc` to `babel.config.js`, is this option still required?

Reply:
Rename `.babelrc` to `babel.config.js`, like https://github.com/vuejs/vue-cli/blob/dev/packages/%40vue/cli-ui-addon-widgets/babel.config.js.  It depends on `process.cwd()`
> The primary downside of this project-wide config is that, because it relies on the working directory, it can be more painful to use in monorepos if the working directory is not the monorepo root. [ref](https://babeljs.io/docs/en/config-files#project-wide-configuration)

If executing `yarn lint` in project root directory,  `@babel/eslint-parser` specified in `cli-ui-addon-widgets/.eslintrc.js` will try to load `process.cwd()/babel.config.js`(Failed with error message 'no babel config ...' ).
>     "lint": "eslint --fix packages/**/*.js packages/**/bin/*",

 If executing `yarn lint` in `packages/@vue/cli-ui-addon-widgets` ,  `@babel/eslint-parser` specified in `cli-ui-addon-widgets/.eslintrc.js` will try to load `process.cwd()/babel.config.js`( `packages/@vue/cli-ui-addon-widget/babel.config.js`) based on `process.cwd()` (Success)
>     "lint": "vue-cli-service lint",

---

If using `babel.config.js` in sub-package, `cwd/root` must be set. No matter executing `yarn lint` in project root directory or sub-package directory, config will be loaded according to `cwd/root` we set.
 
If using `.babelrc` in sub-package and executing `eslint --fix packages/**/*.js packages/**/bin/*`  in project root.
- set `cwd/root` (won't merge with monorepo projectRoot `babel.config.js`) 
- set `babelrcRoot`(will merge  with monorepo projectRoot `babel.config.js` if it exists)
- do not set config above, `.babelrc` will be ignored, `@babel/eslint-parser` can't find `babel.config.js` in project root, emit error messages.
> .babelrc.json files in packages that aren't Babel's 'root' **are ignored unless you opt in with "babelrcRoots"**. [ref](https://babeljs.io/docs/en/config-files#file-relative-configuration)

Since we don't have root babel.config.js,  no matter we use `babel.config.js` or `.babelrc` in sub-package, `cwd`/`root` can work.

All above is how @babel/eslint-parser load babelConfig, it's same with how babel load its config. We should set `cwd`/`root` because we will run ` eslint --fix packages/**/*.js packages/**/bin/*` in project root. When developing sub-package, we will only run `yarn serve` and `yarn build` in sub-package directory and babel-loader will load config as expected.

A little complicated about babel project-wide/file-relative config.

Reply:
Yeah quite complicated… Thanks for the explanation!